easybuild/framework/easyblock.py

```diff
@@ -3385,6 +3385,27 @@ def sanity_check_cuda(self, cuda_dirs=None):
             self.log.info("Using configured subdirectories for binaries/libraries to verify CUDA device code: %s",
                           cuda_dirs)
 
+        # collect all files to consider
+        files_to_check = []
+        for dirpath in [os.path.join(self.installdir, d) for d in cuda_dirs]:
+            if os.path.exists(dirpath):
+                self.log.debug(f"Sanity checking files for CUDA device code under directory {dirpath}:")
+                for entry in os.listdir(dirpath):
+                    path = os.path.join(dirpath, entry)
+                    if os.path.isfile(path):
+                        self.log.debug("Sanity checking file {path} for CUDA device code")
+                        files_to_check.append(path)
+            else:
+                self.log.debug(f"Not sanity checking files in non-existing directory {dirpath}")
+
+        # also consider compiled Python modules as shared libraries (*.so) under lib/python*/site-packages
+        python_pkgs_path = os.path.join(self.installdir, 'lib', 'python*', 'site-packages')
```

Comment:
```suggestion
        python_pkgs_path = os.path.join(self.installdir, 'lib', 'python*', 'site-packages')
```
Is `lib` guaranteed? Is there a way we can check which python paths are being added and use them directly?

Reply:
Either `lib` is a symlink to `lib64`, or the other way around.

We also use `lib` in `EasyBlock.make_module_pythonpath`, so should be fine.

Reply:
That's only true if only one of these exist.